core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -313,27 +315,65 @@ private HttpResponse convertException(ServiceRequestContext ctx, HttpRequest req
      * Creates a {@link Parameter} instance which describes a parameter of the annotated method. If the
      * parameter type is {@link Optional}, its actual argument's type is used.
      */
-    private static Parameter createOptionalSupportedParam(java.lang.reflect.Parameter parameterInfo,
+    private static Parameter createHttpComponentParameter(java.lang.reflect.Parameter parameterInfo,
                                                           ParameterType paramType, String paramValue) {
-        final Default aDefault = parameterInfo.getAnnotation(Default.class);
-        final boolean isOptionalType = parameterInfo.getType() == Optional.class;
+        assert paramType == ParameterType.PARAM ||
+               paramType == ParameterType.HEADER;
```

Comment:
How about appending ` : String.valueOf(paramType)`?

Reply:
Sorry, I can't see what you mean. :-( Would you explain why you want to append that?